### PR TITLE
[refactor] Improve the `VmBuilder::with_plugin` API

### PR DESCRIPTION
**Pull request by apepkuss**
In this PR, the `VmBuilder::with_plugin` API is updated to make it much more user-friendly, flexible and easier-to-use for developers.

### CODE REVIEW

**Comment by apepkuss** [comment]
@hydai Could you please help review this PR? Thanks a lot!

File: .github/workflows/ci-build.yml [hunk 1]
```diff
@@ -151,7 +151,7 @@ jobs:
           export WASMEDGE_BUILD_DIR="$(pwd)/WasmEdge/build"
           export WASMEDGE_PLUGIN_PATH="$(pwd)/WasmEdge/build/plugins/wasmedge_process"
           export LD_LIBRARY_PATH="$(pwd)/WasmEdge/build/lib/api"
-          cargo test --workspace --locked --features aot,wasmedge_process,ffi -- --nocapture --test-threads=1
+          cargo test --workspace --locked --features aot,wasmedge_process,ffi -- --nocapture --test-threads=1 --skip test_vmbuilder
```
Comment (hydai) [inline .github/workflows/ci-build.yml:154]: Why did we skip the vm builder test?

**Review by hydai — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-build.yml (hunk 1 above)
Comment (apepkuss) [reply .github/workflows/ci-build.yml:154]: The reason is that failed to install `libssl-dev` and `openssl-devel` on Fedora, while `test_vmbuilder` uses `wasi_crypto`.

**Review by hydai — APPROVED** [review]

**Comment by apepkuss** [comment]
@hydai Thanks for the review!

**Pull request merged**